frontend/app/src/store/session/index.ts

```diff
@@ -53,9 +59,8 @@ export const useSessionStore = defineStore('session', () => {
       return { success: true };
     } catch (e: any) {
       logger.error(e);
-      if (e instanceof SyncConflictError) {
-        set(syncConflict, { message: e.message, payload: e.payload });
-        return { success: false, message: '' };
+      if (e instanceof SyncConflictError || e instanceof HalfUpgradeError) {
+        return handleLoginError(e);
       }
```

Comment:
This way we do multiple instance of right? can't we make a handler that returns `ActionStatus | null` and only do the instance of inside the method?